src/js/pages/system/OverviewDetailTab.js

```diff
@@ -179,6 +188,101 @@ class OverviewDetailTab extends mixin(StoreMixin) {
     return publicIP;
   }
 
+  /**
+   * Get user if existent otherwise
+   * return null
+   *
+   * @returns {Component|Null} User string or null
+   *
+   * @memberOf OverviewDetailTab
+   */
+  getMesosBuildUser() {
+    const mesosConfig = MesosStateStore.get("lastMesosState");
```

Comment:
It feels wrong to fetch this data once again after we've got it in `getMesosDetails` which is _the only_ function that calls `getMesosBuildUser` so my suggestion would be to pass `mesosConfig` or even `mesosConfig.build_user` as an argument while keeping the logic in `getMesosBuildUser`. 

To you argument about unit testing it would be much easier to test a function that depends on one argument rather that a function that has to mock a Store.

If my arguments aren't strong enough feel free to leave it as it is and merge.